gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapClient.java

```diff
@@ -42,7 +47,9 @@ public UserEntryMapper(String memberOfAttr) {
 
     @Override
     public UserRecord map(Entry entry) throws LdapException {
-      return new UserRecord(entry.get(memberOf).toString());
+      Attribute attribute = entry.get(memberOf);
+      return ObjectUtils.isEmpty(attribute) ? emptyRecord : new UserRecord(attribute.getString());
```

Comment:
Is it possible to change the signature to `Optional<UserRecord>` and return `Optional.empty()` if there is no `memberOf` attribute? `Optional.empty()` is the usual way this type of situation is handled in Trino.